crates/re_viewer_context/src/annotations.rs

```diff
@@ -143,13 +143,19 @@ pub struct ResolvedAnnotationInfo {
 
 impl ResolvedAnnotationInfo {
     /// `rgba` should be unmultiplied
+    #[inline]
     pub fn color(
         &self,
-        rgba: Option<&[u8; 4]>,
+        rgba: Option<[u8; 4]>,
         default_color: DefaultColor<'_>,
     ) -> re_renderer::Color32 {
         if let Some([r, g, b, a]) = rgba {
-            re_renderer::Color32::from_rgba_unmultiplied(*r, *g, *b, *a)
+            if a == 255 {
+                // Common-case optimization
+                re_renderer::Color32::from_rgb(r, g, b)
+            } else {
+                re_renderer::Color32::from_rgba_unmultiplied(r, g, b, a)
+            }
```

Comment:
now that you did https://github.com/emilk/egui/commit/38b4234c3282a7c044c18b77234ee8c204efe171 in egui and given that `from_rgba_unmultiplied` already has this short-circuit, this shouldn't actually be here, amiright? :)

Reply:
It depends on when we update to egui, and wether or not the compiler will actually inline it, and what effect that will have, yadda yadda